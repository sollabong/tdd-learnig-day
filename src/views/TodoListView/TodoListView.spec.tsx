
import React from 'react';
import {TodoListView} from './index'
import { fireEvent, render, screen } from "@testing-library/react";
// describe("<TodoListView />", () => {
//     it("should render view", () => {
//         throw Error("Not implemented");
//     });
// });

describe("<TodoListView />", () => {
    it("should render view", () => {
      render(<TodoListView />);
      expect(screen.getByTestId("todo-list-view")).toMatchSnapshot();
      expect(screen.getByTestId("todo-list-view")).toHaveTextContent("Todo List View");
    });
    it("should render todo input field", () => {
      render(<TodoListView />);
      expect(screen.getByTestId("todo-input")).toBeInTheDocument();
    });
    it("should render button", () => {
      render(<TodoListView />);
      expect(screen.getByTestId("submit-button")).toBeInTheDocument();
    });
    it("should render list", () => {
      render(<TodoListView />);
      expect(screen.getByTestId("todo-list")).toBeInTheDocument();
    });

    describe("the Submit Button", () => {
        it("should update todos state on click", () => {
        });
        // it("add a new element to the todos state", () => {
        //     const { todos } = TodoListView();
        //     // const todosLength = todos.length;
        //     render(<TodoListView />);
        //     const addButton = screen.getByTestId("submit-button");
        //     fireEvent.click(addButton);
        //     expect(todos).toContain()
        //   });
    })
});