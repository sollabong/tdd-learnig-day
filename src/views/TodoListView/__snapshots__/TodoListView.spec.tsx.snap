// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TodoListView /> should render view 1`] = `
<div
  data-testid="todo-list-view"
>
  <h1>
    Todo List View
  </h1>
  <input
    data-testid="todo-input"
    type="text"
  />
  <button
    data-testid="submit-button"
  >
    Submit
  </button>
  <div
    data-testid="todo-list"
  />
</div>
`;
